.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [cicd]
+  pull_request:
+    branches: [master]
```

Comment:
given that the below is just checking stuff (eslint, prettier, tests), won't we want to also run this for `develop`? I never really used `master` for anything fwiw

PS: we should rename master -> main at some poin if keeping

Reply:
Sure! My high level plan is to eventually have three core branches:
 - main (which deploys to npm)
 - staging (which deploys to the demo)
 - dev (which deploys to some dev environment which currently doesn't exist)

I'll update the config file to reflect this

Reply:
Why not just one branch for everything? Like just `develop` and we drop all the rest?

AFAIK the npm deploy won't happen if the patch/minor version hasn't changed, so we could continue to use the packages semvar/version for versioning and controlling when things get "deployed" to npm. For dev I feel we could continue to use `develop` and just CR the changes

I know this is controversial but I feel like the best branching strategy in git is no branching strategy, just doing everything in one branch (I googled and I guess this is called the "trunk" strategy: https://launchdarkly.com/blog/introduction-to-trunk-based-development/). I've seen it work for some massive, massive projects. I've seen a multi-branch strategies be a PITA for even small projects

Reply:
I guess, phrased another way, what will multiple branches unlock that we can't do with one? As far as I can tell, we can  do the three things listed with a single `develop` branch -- and have done so. So we might as well keep things simple and just leave `develop`

re: only doing the npm deploy if the package's version has changed within a GitHub, we should be able to check the local version vs what's in npm: https://stackoverflow.com/a/26174890

Reply:
I'm not very familiar with the trunk strategy, I'll look into it.

I like having multiple branches as I often want to deploy to different environments. In my experience it's been a nice way to deploy to dev, staging, and prod environments via a CICD pipeline. To mitigate some of the complexity of the branching strategy I like to freeze prod and only allow merges (gated by review).

In any case, I'm not really beholden to one style or the other. How do you envision setting up automated deployments to three separate environments using one branch?

Reply:
Re: dev, staging and prod environments, we could still accomplish that with a single branch -- it's just that a single commit/change of that branch works its way thru each env (and each env's corresponding approvals).

```
         dev     ->     staging     ->   prod
         c1
         c1      ->      c1 (bug found in staging w/ c1)
         c2      ->     c2           ->     c2
```

I'm familiar with the multi-branch strategy as a means of separating what gets deployed to where, but ofc it adds the additional complexity of someone needing to keep track of the state of 3 branches vs one. For what seqviz is (a UI library) I do think that that we should bias towards simplicity. Ie we have a single branch and track differences with any (or a combination of):
- commit hashes
- tages
- package version

Fwiw though I'm also not glued to one approach or other. If you think that this will unlock something we couldn't do easily with one branch, I'm game